#### Set WD & Load Libraries ####
getwd() # use setwd("path/to/files") if you are not in the right directory
#setwd("/Volumes/HLF_SSD/Aronson_Lab_Data/Salton_Sea/SaltonSeaWater")
suppressPackageStartupMessages({ # load packages quietly
  library(devtools)
  library(phyloseq)
  library(ggplot2)
  library(vegan)
  library(ggpubr)
  library(lme4)
  #library(scales)
  library(grid)
  library(ape)
  library(plyr)
  library(dplyr)
  library(viridis)
  library(ggbiplot)
  library(readxl)
  library(metagenomeSeq)
  library(DESeq2)
  library(dplyr)
  library(magrittr)
  library(MASS)
  library(dendextend)
  library(tidyr)
  library(reshape)
  library(reshape2)
  library(wesanderson)
  library(nationalparkcolors)
  library(shades)
  library(ALDEx2)
  library(rstatix)
  library(decontam)
  library(ggvegan)
  library(microbiome)
  library(pairwiseAdonis)
  library(corrplot)
})

#### Load Global Env to Import Count/ASV Tables ####
load("data/SSeawater_Data_Ready.Rdata") # save global env to Rdata file
load("data/SSeawater_AlphaDiv_Data_Rarefied.Rdata")
#load("data/ssw_clr.euc.dist_2.21.23.Rdata")

#save.image("data/Env_Seqs_All/env.seq_analysis.Rdata") # save global env to Rdata file
bac.dat.all[1:6,1:6]
bac.ASV_table[1:4,1:4]
bac.ASV_table[(nrow(bac.ASV_table)-4):(nrow(bac.ASV_table)),(ncol(bac.ASV_table)-4):(ncol(bac.ASV_table))] # last 4 rows & cols
head(meta_scaled)

## DO NOT RUN THIS LINE, THIS IS YOUR COLOR REFERENCE!!!!
#(August.2021="#ef781c",December.2021="#03045e",April.2022="#059c3f")

#### Rarefaction Curves & Species Accumulation Curves ####
# bacteria/archaea

# Species Accumulation Curve
sc2<-specaccum(bac.ASV_table[,-1],"random")
plot(sc2, ci.type="poly", col="darkgreen", lwd=2, ci.lty=0, ci.col="lightgreen")
boxplot(sc2, col="yellow", add=TRUE, pch=20)

# Prep for Rarefaction Curve
rowSums(bac.ASV_table[,-1]) # total # ASVs per sample, excluding SampleID from calculation
sort(colSums(bac.ASV_table[,-1]))

# Create Rarefaction curve
png('figures/AlphaDiversity/SSW_16S_rarecurve.png')
rarecurve(as.matrix(bac.ASV_table[,-1]),col=metadata$SampDate_Color, step=1000, label=F,ylab="ASVs")
# to show sampel labels per curve, change label=T
dev.off()

#### Good's Coverage ####

# Good's coverage is defined as 1 - (F1/N) where F1 is the number of singleton OTUs and N is the sum of counts for all OTUs.
# If a sample has a Good's coverage == . 98, this means that 2% of reads in that sample are from OTUs that appear only once in that sample.
# NOTE: Good's Coverage is not the best estimate of coverage because DADA2 removes singletons during process - so we are basing this on singeltons that remain (??)
bac.ASV.melt<-melt(bac.ASV_table,by="SampleID")
colnames(bac.ASV.melt)[which(colnames(bac.ASV.melt) == "variable")] <- "ASV_ID"
colnames(bac.ASV.melt)[which(colnames(bac.ASV.melt) == "value")] <- "Count"

# calculate # of singletons per sample
sings<-data.frame(Singletons=rowSums(bac.ASV_table[,-1]==1),SampleID=bac.ASV_table$SampleID) # how many ASVs have a count of only 1

# find total counts per sample
totseq<-data.frame(TotalSeqs=rowSums(bac.ASV_table[,-1]),SampleID=bac.ASV_table$SampleID) # total # of counts per sample

# Calculate Good's Coverage
good.cov<-data.frame(GoodsCov=(100*(1-(sings$Singletons/totseq$TotalSeqs))),SampleID=sings$SampleID)

# Merge Total Seqs and Goods DFs together for plotting
good.df<-merge(good.cov,totseq,by="SampleID")

ggplot(good.df, aes(x=TotalSeqs,y=GoodsCov))+geom_point()+xlab("Total Seqs per Sample")+ylab("Good's Coverage (%)")


#### Rarefaction of Raw Counts ####
# in vegan ROWS need to be SITES/samples; COLUMNS are SPECIES (OTUs, ASVs)
min.rar<-min(rowSums(bac.ASV_table[,-1])) ## seeing min sum of OTUs so we can see what min is for rarefaction
min.rar

bac.ASV.rar<-rrarefy(bac.ASV_table[,-1],min.rar) ## be cognizant of min for rarefaction
bac.ASV.rar[1:4,1:4]

#### Variance Stabilizing Transformation VST of Raw (?) counts ####
#Prepare Contig Feature Count Data for Normalization w/ DESeq2
# make sure count data & mgm_meta are in the same order
bac.ASV_table[1:5,1:5]

#bac.ASV_matrix<-as.matrix(bac.ASV_table[,!names(bac.ASV_table) %in% c("SampleID")]) # convert count table into matrix & exclude column called SampleID
bac.ASV_matrix2<-t(as.matrix(bac.ASV_table[,-1])) # transpose matrix so ASVs are rows, samples are columns
# ^ will be used in DESeq2 functions
rownames(meta_scaled) %in% colnames(bac.ASV_matrix2) # check if rownames in metadata (SampleID) match column names in count data
dim(meta_scaled)
dim(bac.ASV_matrix2)

# Reorder matrix to match order of metadata
bac.ASV_matrix2=bac.ASV_matrix2[,rownames(meta_scaled)] ## reorder ASV matrix by column name to match order of rownames in meta_scaled
colnames(bac.ASV_matrix2) # sanity check that this reordering worked
rownames(meta_scaled)

# create the DESeq DataSet object for DGE analysis
# DESeq2 needs whole # data, so need raw read counts, NOT coverage for these tables...questionable
# b_dds has the scaled coverage that was calculated by dividing reads from featureCounts by gene lengths
b_dds<-DESeqDataSetFromMatrix(countData=round(bac.ASV_matrix2), colData = meta_scaled, design = ~ 1)

# design = ~ 1 means no design
head(counts(b_dds))
colSums(counts(b_dds)) %>% barplot

# Estimate size factor - aka normalization factor, divide all read counts by each size factor per sample
b_dds <- estimateSizeFactors(b_dds,type="ratio")
## To calculate size factor in DESeq2:
# calculates geometric mean of each gene in each sample and uses this as a pseudoreference
# calculates ratio of each sample by dividing each gene count by it's pseudoreference in each sample
# The median value of all ratios for a given sample is taken as the normalization factor (size factor)
# The differentially expressed genes should not affect the median value
# median of ratios method makes the assumption that not ALL genes are differentially expressed; therefore, the normalization factors should account for sequencing depth and RNA composition of the sample
## (large outlier genes will not represent the median ratio values)
# more here: https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html

sizeFactors(b_dds)

plot(sizeFactors(b_dds), colSums(counts(b_dds)))

### Variance Stabilizing Transformation
# you should be able to use matrix or DESeq2 object for this next function, but matrix was not working?
b_vst1 <- varianceStabilizingTransformation(b_dds) # add pseudocount
assay(b_vst1) #see output of VST

b.vst<-assay(b_vst1)

#### Compare Transformed ASVs vs Raw Counts ####
total_asvs<-data.frame(ASV_Total=rowSums(bac.ASV_table[,-1]),metadata)
total_asvs$SampleID = factor(total_asvs$SampleID, levels=unique(total_asvs$SampleID[order(total_asvs$ASV_Total)]), ordered=TRUE)

ggplot(data=total_asvs, aes(x=SampleID, y=ASV_Total,fill=Sample_Type)) +
  geom_bar(stat="identity",colour="black")+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# Calculate rarefied ASVs per Sample
total_rar_asvs<-data.frame(ASV_Total=rowSums(bac.ASV.rar),metadata)

ggplot(data=total_rar_asvs, aes(x=SampleID, y=ASV_Total,fill=Sample_Type)) +
  geom_bar(stat="identity",colour="black")+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# Calculate VST ASVs per Sample
total_vst_asvs<-data.frame(ASV_Total=colSums(b.vst),metadata)
total_vst_asvs$SampleID = factor(total_vst_asvs$SampleID, levels=unique(total_vst_asvs$SampleID[order(total_vst_asvs$ASV_Total)]), ordered=TRUE)

ggplot(data=total_vst_asvs, aes(x=SampleID, y=ASV_Total,fill=Sample_Type)) +
  geom_bar(stat="identity",colour="black")+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

## average ASV per sample month & depth
aggregate(bac.ASV_all$Count, list(bac.ASV_all$SampleMonth), FUN=mean)
aggregate(bac.ASV_all$Count, list(bac.ASV_all$Depth_m), FUN=mean)

#### Alpha Diversity & Species Richness - Rarefied Data ####

## Calculate Shannon Diversity (abundance + richness considered in diversity calculation)
# if you have another package loaded that has a diversity function, you can specify that you want to use vegan's diversity function as shown below
Shan_ent.16s.rar<-vegan::diversity(bac.ASV.rar, index="shannon") # Shannon entropy
Shan_div.16s.rar<- exp(Shan_ent.16s.rar) # Shannon Diversity aka Hill number 1

# create data frame with Shannon entropy and Shannon diversity values
div_16s.rar<-data.frame(Bac_Shannon_Entropy=Shan_ent.16s.rar,Bac_Shannon_Diversity=Shan_div.16s.rar)
class(div_16s.rar)
div_16s.rar$SampleID<-rownames(div_16s.rar)
head(div_16s.rar)

# Calculate species richness (number of species per sample)
specnumber(bac.ASV.rar)

# Create a DF with Species Richness
S_16s.rar<-data.frame(Bac_Species_Richness=specnumber(bac.ASV.rar), SampleID=rownames(bac.ASV.rar)) # finds # of species per sample using RAW count data; if MARGIN = 2 it finds frequencies of species

# merge richness and diversity dataframes together
d.r_16s.rar<-merge(div_16s.rar, S_16s.rar, by.x="SampleID", by.y="SampleID")

# merge w/ metadata
bac.div.metadat.rar <- merge(d.r_16s.rar,meta_scaled, by.x="SampleID", by.y="SampleID")
head(bac.div.metadat.rar)
class(bac.div.metadat.rar) # want data frame

unique(bac.div.metadat.rar$SampleMonth) # see how many elements there are in the Group variable
unique(bac.div.metadat.rar$Depth_m) # see how many elements there are in the Group variable
bac.div.metadat.rar$Depth_m<-factor(bac.div.metadat.rar$Depth_m, levels=c("0","3","4","5","7","9","10","10.5"))

# drop the outliers
#bac.div.metadat.rar<-bac.div.metadat.rar[bac.div.metadat.rar$Bac_Shannon_Diversity<300 & bac.div.metadat.rar$Bac_Species_Richness>100,]

# create numeric variable for depth to be used for models later
bac.div.metadat.rar$Depth.num<-as.numeric(as.character(bac.div.metadat.rar$Depth_m))

# Find highest/lowest values of Shannon div per sample date
max(bac.div.metadat.rar$Bac_Shannon_Diversity[bac.div.metadat.rar$SampDate=="August.2021"]) # max div August 2021
min(bac.div.metadat.rar$Bac_Shannon_Diversity[bac.div.metadat.rar$SampDate=="August.2021"]) # min div August 2021
bac.div.metadat.rar[bac.div.metadat.rar$SampDate=="August.2021",]

max(bac.div.metadat.rar$Bac_Shannon_Diversity[bac.div.metadat.rar$SampDate=="December.2021"]) # max div Dec 21
min(bac.div.metadat.rar$Bac_Shannon_Diversity[bac.div.metadat.rar$SampDate=="December.2021"]) # min div Dec 21
bac.div.metadat.rar[bac.div.metadat.rar$SampDate=="December.2021",]

max(bac.div.metadat.rar$Bac_Shannon_Diversity[bac.div.metadat.rar$SampDate=="April.2022"]) # max div Apr 22
min(bac.div.metadat.rar$Bac_Shannon_Diversity[bac.div.metadat.rar$SampDate=="April.2022"]) # min div Apr 22
bac.div.metadat.rar[bac.div.metadat.rar$SampDate=="April.2022",]

# Find highest/lowest values of Species richness per sample date
max(bac.div.metadat.rar$Bac_Species_Richness[bac.div.metadat.rar$SampDate=="August.2021"]) # max sr August 2021
min(bac.div.metadat.rar$Bac_Species_Richness[bac.div.metadat.rar$SampDate=="August.2021"]) # min sr August 2021
bac.div.metadat.rar[bac.div.metadat.rar$SampDate=="August.2021",]

max(bac.div.metadat.rar$Bac_Species_Richness[bac.div.metadat.rar$SampDate=="December.2021"]) # max sr Dec 2021
min(bac.div.metadat.rar$Bac_Species_Richness[bac.div.metadat.rar$SampDate=="December.2021"]) # min sr Dec 2021
bac.div.metadat.rar[bac.div.metadat.rar$SampDate=="December.2021",]

max(bac.div.metadat.rar$Bac_Species_Richness[bac.div.metadat.rar$SampDate=="April.2022"]) # max sr April 2022
min(bac.div.metadat.rar$Bac_Species_Richness[bac.div.metadat.rar$SampDate=="April.2022"]) # min sr April 2022
bac.div.metadat.rar[bac.div.metadat.rar$SampDate=="April.2022",]

# save diversity data
save.image("data/SSeawater_AlphaDiv_Data_Rarefied.Rdata")

#### Using Shapiro-Wilk test for Normality - Rarefied Data ####
shapiro.test(bac.div.metadat.rar$Bac_Shannon_Diversity) # what is the p-value?
# p-value = 0.8612
# p > 0.05 states distribution of data are not significantly different from normal distribution
# p < 0.05 means that data is significantly different from a normal distribution
hist(bac.div.metadat.rar$Bac_Shannon_Diversity, col="blue") # with outliars

# visualize Q-Q plot for alpha div
# The Q-Q plot, or quantile-quantile plot, is a graphical tool to help us assess if a set of data plausibly came from some theoretical distribution such as a normal or exponential.
# For example, if we run a statistical analysis that assumes our residuals are normally distributed, we can use a normal Q-Q plot to check that assumption
# more on Q-Q plots here: https://data.library.virginia.edu/understanding-q-q-plots/
# more here too: https://grodri.github.io/glms/notes/c2s9#:~:text=8%20The%20Q%2DQ%20Plot,versus%20quantiles%20of%20a%20distribution.
qqnorm(bac.div.metadat.rar$Bac_Shannon_Diversity, pch = 1, frame = FALSE)
qqline(bac.div.metadat.rar$Bac_Shannon_Diversity, col = "red", lwd = 2)

shapiro.test(bac.div.metadat.rar$Bac_Species_Richness) # what is the p-value?
# p-value = 0.009533
# p > 0.05 states distribution of data are not significantly different from normal distribution
# p < 0.05 means that data is significantly different from a normal distribution
hist(bac.div.metadat.rar$Bac_Species_Richness, col="blue")

# visualize Q-Q plot for species richness
qqnorm(bac.div.metadat.rar$Bac_Species_Richness, pch = 1, frame = FALSE) # with outliars
qqline(bac.div.metadat.rar$Bac_Species_Richness, col = "red", lwd = 2)

qqnorm(bac.div.metadat.rar$Bac_Species_Richness, pch = 1, frame = FALSE) # without outliars
qqline(bac.div.metadat.rar$Bac_Species_Richness, col = "red", lwd = 2)

shapiro.test(bac.div.metadat.rar$DO_Percent_Local) # p-value = 0.02586
hist(bac.div.metadat.rar$DO_Percent_Local, col="blue")
# visualize Q-Q plot for species richness
qqnorm(bac.div.metadat.rar$DO_Percent_Local, pch = 1, frame = FALSE) # with outliars
qqline(bac.div.metadat.rar$DO_Percent_Local, col = "red", lwd = 2)

shapiro.test(bac.div.metadat.rar$ORP_mV) # p-value = 1.731e-08
hist(bac.div.metadat.rar$ORP_mV, col="blue")
# visualize Q-Q plot for species richness
qqnorm(bac.div.metadat.rar$ORP_mV, pch = 1, frame = FALSE) # with outliars
qqline(bac.div.metadat.rar$ORP_mV, col = "red", lwd = 2)

shapiro.test(bac.div.metadat.rar$Temp_DegC) # p-value = 0.0002829
hist(bac.div.metadat.rar$Temp_DegC, col="blue")
# visualize Q-Q plot for species richness
qqnorm(bac.div.metadat.rar$Temp_DegC, pch = 1, frame = FALSE) # with outliars
qqline(bac.div.metadat.rar$Temp_DegC, col = "red", lwd = 2)

shapiro.test(bac.div.metadat.rar$Dissolved_OrganicMatter_RFU) #  p-value = 0.05411
hist(bac.div.metadat.rar$Dissolved_OrganicMatter_RFU, col="blue")
# visualize Q-Q plot for species richness
qqnorm(bac.div.metadat.rar$Dissolved_OrganicMatter_RFU, pch = 1, frame = FALSE) # with outliars
qqline(bac.div.metadat.rar$Dissolved_OrganicMatter_RFU, col = "red", lwd = 2)

shapiro.test(bac.div.metadat.rar$Sulfate_milliM) # p-value = 0.1912
hist(bac.div.metadat.rar$Sulfate_milliM, col="blue")
# visualize Q-Q plot for species richness
qqnorm(bac.div.metadat.rar$Sulfate_milliM, pch = 1, frame = FALSE) # with outliars
qqline(bac.div.metadat.rar$Sulfate_milliM, col = "red", lwd = 2)

shapiro.test(bac.div.metadat.rar$Sulfide_microM) # p-value = 3.813e-08
hist(bac.div.metadat.rar$Sulfide_microM, col="blue")
# visualize Q-Q plot for species richness
qqnorm(bac.div.metadat.rar$Sulfide_microM, pch = 1, frame = FALSE) # with outliars
qqline(bac.div.metadat.rar$Sulfide_microM, col = "red", lwd = 2)

#### Visualize Alpha Diversity & Species Richness - from Rarefied Data ####
## Shannon Diversity by Sample Month & Depth
bac.a.div.rar<-ggplot(bac.div.metadat.rar, aes(x=SampDate, y=Bac_Shannon_Diversity)) +geom_jitter(aes(color=as.numeric(as.character(Depth_m))), size=3, width=0.15, height=0) +
  scale_colour_gradient2(low="red",high="blue3",midpoint=5,guide = guide_colourbar(reverse = TRUE)) +
  geom_boxplot(fill=NA, outlier.color=NA)+scale_x_discrete(labels=c("August 2021","December 2021","April 2022"))+theme_bw()+theme_classic()+
  labs(title = "Bacterial Shannon Diversity by Sample Date & Depth", subtitle="Using Rarefied Counts", x="Sample Date", y="Shannon Diversity", color="Depth (m)")+theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1,size=10),legend.title.align=0.5, legend.title = element_text(size=13),legend.text = element_text(size=11),plot.title = element_text(size=15)) +
  geom_pwc(method = "t_test", label = "p.adj.format",p.adjust.method = "bonferroni")

ggsave(bac.a.div.rar,filename = "figures/AlphaDiversity/RarefiedCounts/SSW_16S_rarefied_alpha_diversity_sampledate_depth_boxplot.png", width=13, height=10, dpi=600)

bac.div.metadat.rar$Depth_m=as.numeric(levels(bac.div.metadat.rar$Depth_m))[bac.div.metadat.rar$Depth_m]
# ^ note: cannot turn numbers that are factors in R into numeric values...
## have to convert factor levels into numeric, then use the numeric "levels" to pull out numbers from Depth_m column in df to make sure the Depth_m columns is now numeric, not a factor

# bac.a.div.rar2<-ggplot(bac.div.metadat.rar, aes(x=as.factor(Depth_m), y=Bac_Shannon_Diversity)) +geom_boxplot(aes(fill=Depth_m),color="black")+
#   labs(title = "Bacterial Shannon Diversity by Sampling Depth", x="Depth (m)", y="Shannon Diversity", fill="Depth (m)")+
#   scale_fill_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) + theme_classic() +
#   theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1,,size=10),legend.title.align=0.5, legend.title = element_text(size=13),legend.text = element_text(size=11),plot.title = element_text(size=15)) +
#   coord_flip() + scale_x_discrete(limits=rev)
#
# ggsave(bac.a.div2,filename = "figures/AlphaDiversity/RarefiedCounts/SSW_Bacterial_rarefied_alpha_diversity_depth_boxplot_v1.png", width=13, height=10, dpi=600)
#
# bac.a.div3<-ggplot(bac.div.metadat.rar, aes(x=as.factor(Depth_m), y=Bac_Shannon_Diversity)) +geom_boxplot(aes(fill=Depth_m),color="black")+
#   labs(title = "Bacterial Shannon Diversity by Sampling Depth", x="Depth (m)", y="Shannon Diversity", fill="Depth (m)")+
#   scale_fill_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) + theme_classic() +
#   theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1,,size=10),legend.title.align=0.5, legend.title = element_text(size=13),legend.text = element_text(size=11),plot.title = element_text(size=15))
#
# ggsave(bac.a.div3,filename = "figures/AlphaDiversity/RarefiedCounts/SSW_Bacterial_rarefied_alpha_diversity_depth_boxplot_v2.png", width=13, height=10, dpi=600)

## Species Richness by Sample Type
bac.a.sr.rar<-ggplot(bac.div.metadat.rar, aes(x=SampDate, y=Bac_Species_Richness)) +geom_jitter(aes(color=as.numeric(as.character(Depth_m))), size=3, width=0.15, height=0) +
  scale_colour_gradient2(low="red",high="blue3",midpoint=5,guide = guide_colourbar(reverse = TRUE)) +
  geom_boxplot(fill=NA, outlier.color=NA)+scale_x_discrete(labels=c("August 2021","December 2021","April 2022"))+theme_bw()+theme_classic()+
  labs(title = "Bacterial Species Richness by Sample Date & Depth", subtitle="Using Rarefied Counts", x="Sample Date", y="Species Richness", color="Depth (m)")+theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1,size=10),legend.title.align=0.5, legend.title = element_text(size=13),legend.text = element_text(size=11),plot.title = element_text(size=15)) +
  stat_compare_means(method="anova")

ggsave(bac.a.sr.rar,filename = "figures/AlphaDiversity/RarefiedCounts/SSW_Bacterial_rarefied_species_richness_samplemonth_depth_boxplot.png", width=13, height=10, dpi=600)

# bac.a.sr.rar2<-ggplot(bac.div.metadat.rar, aes(x=as.factor(Depth_m), y=Bac_Species_Richness,fill=bac.div.metadat.rar$Depth_m)) +geom_boxplot(aes(fill=as.numeric(bac.div.metadat.rar$Depth_m)),color="black")+
#   labs(title = "Bacterial Species Richness by Sampling Depth", x="Depth (m)", y="Species Richness", fill="Depth (m)")+
#   scale_fill_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) + theme_classic() +
#   theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1,,size=10),legend.title.align=0.5, legend.title = element_text(size=13),legend.text = element_text(size=11),plot.title = element_text(size=15)) +
#   coord_flip() + scale_x_discrete(limits=rev)
#
# ggsave(bac.a.sr.rar2,filename = "figures/AlphaDiversity/RarefiedCounts/SSW_Bacterial_rarefied_species_richness_depth_boxplot_v1.png", width=13, height=10, dpi=600)
#
# bac.a.sr.rar3<-ggplot(bac.div.metadat.rar, aes(x=as.factor(Depth_m), y=Bac_Species_Richness,fill=Depth_m)) +geom_boxplot(aes(fill=as.numeric(bac.div.metadat.rar$Depth_m)),color="black")+
#   labs(title = "Bacterial Species Richness by Sampling Depth", x="Depth (m)", y="Species Richness", fill="Depth (m)")+
#   scale_fill_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) + theme_classic() +
#   theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1,,size=10),legend.title.align=0.5, legend.title = element_text(size=13),legend.text = element_text(size=11),plot.title = element_text(size=15))
#
# ggsave(bac.a.sr.rar3,filename = "figures/AlphaDiversity/RarefiedCounts/SSW_Bacterial_rarefied_species_richness_depth_boxplot_v2.png", width=13, height=10, dpi=600)



#### Linear Regression/ANOVA Comparisons - Shannon Diversity ####
## here the focus is comparing env variables of interest to see if they can predict diversity and richness
head(bac.div.metadat.rar)

# just look at everything at once in step-wise fashion
step1<-step(glm(formula = Bac_Shannon_Diversity ~ ., data=bac.div.metadat.rar[,c(3,11,13:14,18:20)]))
summary(step1)
# Coefficients:
#   Estimate Std. Error t value Pr(>|t|)
# (Intercept)      86.842      3.002  28.926  < 2e-16 ***
#   ORP_mV           -6.478      3.664  -1.768 0.092285 .
# Temp_DegC       -21.212      4.471  -4.744 0.000124 ***
#   Sulfate_milliM  -10.994      3.886  -2.829 0.010374 *

div.glm.all<-glm(formula = Bac_Shannon_Diversity ~ ., data=bac.div.metadat.rar[,c(3,11,13:14,18:20)])
summary(div.glm.all)

div.glm.p<-coef(summary(div.glm.all))[,4] # p-values
Div.GLM.Pval<-data.frame(Div.GLM.AdjPval=p.adjust(div.glm.p, method="bonferroni",n=length(div.glm.p)),Div.GLM.Pval=div.glm.p)
#                               Div.GLM.AdjPval Div.GLM.Pval
# (Intercept)                    8.353610e-15 1.193373e-15
# DO_Percent_Local               1.000000e+00 6.614790e-01
# ORP_mV                         1.000000e+00 3.249102e-01
# Temp_DegC                      4.793721e-02 6.848173e-03 *
# Dissolved_OrganicMatter_RFU    1.000000e+00 6.835361e-01
# Sulfate_milliM                 2.762433e-01 3.946333e-02
# Sulfide_microM                 1.000000e+00 6.239275e-01

# [Example Code for Different Models]

s.div.glm.fit1<-glm(formula = Bac_Shannon_Diversity ~ DO_Percent_Local, data=bac.div.metadat.rar)%>%
  adjust_pvalue(method="bonferroni")

# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(s.div.glm.fit1)

# sanity check that lm() vs glm(familiy=Gaussian) is the same thing - and it is!
summary(lm(formula = Bac_Shannon_Diversity ~ DO_Percent_Local, data=bac.div.metadat.rar)%>%
          adjust_pvalue(method="bonferroni"))

# code for mixed effects model
mixed1 = lmer(Bac_Shannon_Diversity ~ DO_Percent_Local+ (1 | Depth_m), data = bac.div.metadat.rar)
summary(mixed1)

# Shan Div ~ DO%
plot(Bac_Shannon_Diversity ~ DO_Percent_Local, data=bac.div.metadat.rar,col=SampDate_Color)

# Shan Div ~ ORP

plot(Bac_Shannon_Diversity ~ ORP_mV, data=bac.div.metadat.rar,col=SampDate_Color)

# Shan Div ~ Temp (C)

plot(Bac_Shannon_Diversity ~ Temp_DegC, data=bac.div.metadat.rar,col=SampDate_Color)

# Shan Div ~ DOM

plot(Bac_Shannon_Diversity ~ Dissolved_OrganicMatter_RFU, data=bac.div.metadat.rar,col=SampDate_Color)

# Shan Div ~ Sulfate

plot(Bac_Shannon_Diversity ~ Sulfate_milliM, data=bac.div.metadat.rar,col=SampDate_Color)

# Shan Div ~ Sulfide

plot(Bac_Shannon_Diversity ~ Sulfide_microM, data=bac.div.metadat.rar,col=SampDate_Color)

# Shan Div ~ Depth
plot(Bac_Shannon_Diversity ~ Depth.num, data=bac.div.metadat.rar,col=SampDate_Color)

fit1<-aov(Bac_Shannon_Diversity ~ SampDate, data=bac.div.metadat.rar)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(bac.div.metadat.rar$Bac_Shannon_Diversity, bac.div.metadat.rar$SampDate, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit1)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
#SampDate     2   3648  1824.0   6.677 0.0057 **
# Residuals   21   5737   273.2

p.adjust(summary(fit1)[[1]][["Pr(>F)"]][1],method="bonferroni")

# Tukey test - tells us which groups are significantly different from each other (more here: https://www.r-bloggers.com/2013/06/anova-and-tukeys-test-on-r/)
Tuk1<-TukeyHSD(fit1)
Tuk1$SampDate
#                             diff        lwr      upr      p adj
# December.2021-August.2021 25.076433   4.245680 45.90719 0.016639615 *
# April.2022-August.2021    27.111615   6.280862 47.94237 0.009571804 **
# April.2022-December.2021   2.035182 -18.795571 22.86593 0.967173383

# fit.0<-aov(DustComplexity ~ as.factor(Elevation), data=bac.div.metadat.rar)
# summary(fit.0)
# TukeyHSD(fit.0)
# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Shannon_Diversity ~ SampDate, data = bac.div.metadat.rar)
# Fligner-Killeen:med chi-squared = 1.9504, df = 2, p-value = 0.3771
# Which shows that the data do not deviate significantly from homogeneity.
compare_means(Bac_Shannon_Diversity ~ SampDate, data=bac.div.metadat.rar, method="anova",p.adjust.method = "bonferroni") # won't take as.factor(Elevation) as input

#### Linear Regression/ANOVA Comparisons - Species Richness ####
## here the focus is comparing dust complexity to alpha diversity, species richness, & elevation
head(bac.div.metadat.rar) # bac.div.metadat.rar - excludes outliar with very high Shannon diversity

# just look at everything at once in step-wise fashion
step2<-step(glm(formula = Bac_Species_Richness ~ ., data=bac.div.metadat.rar[,c(4,11,13:14,18:20)]))
summary(step2)
#                               Estimate Std. Error t value Pr(>|t|)
# (Intercept)      720.46      19.00  37.928   <2e-16 ***
# ORP_mV          -234.80      95.83  -2.450   0.0236 *
# Sulfate_milliM   -33.65      19.91  -1.690   0.1066
# Sulfide_microM  -222.28      96.33  -2.307   0.0319 *

sr.glm.all<-glm(formula = Bac_Species_Richness ~ ., data=bac.div.metadat.rar[,c(4,11,13:14,18:20)])
summary(sr.glm.all)

sr.glm.p<-coef(summary(sr.glm.all))[,4] # p-values
SR.GLM.Pval<-data.frame(SR.GLM.AdjPval=p.adjust(sr.glm.p, method="bonferroni",n=length(sr.glm.p)),SR.GLM.Pval=sr.glm.p)
#                               SR.GLM.AdjPval  SR.GLM.Pval
# (Intercept)                   9.032213e-17 1.290316e-17
# DO_Percent_Local              1.000000e+00 6.008282e-01
# ORP_mV                        4.753271e-01 6.790388e-02
# Temp_DegC                     1.000000e+00 3.950656e-01
# Dissolved_OrganicMatter_RFU   1.000000e+00 3.652658e-01
# Sulfate_milliM                5.244600e-01 7.492285e-02
# Sulfide_microM                8.091316e-01 1.155902e-01

# Species Richness ~ DO%

plot(Bac_Species_Richness ~ DO_Percent_Local, data=bac.div.metadat.rar,col=SampDate_Color)

# Species Richness ~ ORP

plot(Bac_Species_Richness ~ ORP_mV, data=bac.div.metadat.rar,col=SampDate_Color)

# Species Richness ~ Temp

plot(Bac_Species_Richness ~ Temp_DegC, data=bac.div.metadat.rar,col=SampDate_Color)

# Species Richness ~ DOM

plot(Bac_Species_Richness ~ Dissolved_OrganicMatter_RFU, data=bac.div.metadat.rar,col=SampDate_Color)

# Species Richness ~ Sulfate

plot(Bac_Species_Richness ~ Sulfate_milliM, data=bac.div.metadat.rar,col=SampDate_Color)

# Species Richness ~ Sulfide

plot(Bac_Species_Richness ~ Sulfide_microM, data=bac.div.metadat.rar,col=SampDate_Color)

# Species Richness ~ Depth

plot(Bac_Species_Richness ~ Depth.num, data=bac.div.metadat.rar,col=SampDate_Color)

fit2<-aov(Bac_Species_Richness ~ SampDate, data=bac.div.metadat.rar)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(bac.div.metadat.rar$Bac_Species_Richness, bac.div.metadat.rar$Depth_m, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit2)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
# SampDate     2   1783     891   0.077  0.926
# Residuals   21 242367   11541
p.adjust(summary(fit2)[[1]][["Pr(>F)"]][1],method="bonferroni")

# Tukey test - tells us which groups are significantly different from each other (more here: https://www.r-bloggers.com/2013/06/anova-and-tukeys-test-on-r/)
Tuk2<-TukeyHSD(fit2)
Tuk2$SampDate
#                               diff       lwr       upr       p adj
# December.2021-August.2021 -18.875 -154.268 116.518 0.9344133
# April.2022-August.2021    -17.625 -153.018 117.768 0.9425413
# April.2022-December.2021    1.250 -134.143 136.643 0.9997015

# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Species_Richness ~ SampDate, data = bac.div.metadat.rar)
# Fligner-Killeen:med chi-squared = 6.1181, df = 2, p-value = 0.04693
# Which shows that the data DO deviate significantly from homogeneity.
compare_means(Bac_Species_Richness ~ SampDate, data=bac.div.metadat.rar, method="anova",p.adjust.method = "bonferroni")

#### Prep Data for Linear Regressions within Timepoints ####
## here the focus is comparing dust complexity to alpha diversity, species richness, & elevation
head(bac.div.metadat.rar)

# create the dataframes
aug21.div<-subset(bac.div.metadat.rar, bac.div.metadat.rar$SampDate=="August.2021")
dec21.div<-subset(bac.div.metadat.rar, bac.div.metadat.rar$SampDate=="December.2021")
apr22.div<-subset(bac.div.metadat.rar, bac.div.metadat.rar$SampDate=="April.2022")

#### August - Shannon Diversity ####
# August 2021
aug21.div.glm.fit1<-glm(formula = Bac_Shannon_Diversity ~ DO_Percent_Local, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.div.glm.fit1)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)       0.0112706  0.0004664  24.166   <2e-16 ***
#DO_Percent_Local -0.0009547  0.0005092  -1.875   0.0687 .

aug21.div.glm.fit2<-glm(formula = Bac_Shannon_Diversity ~ ORP_mV, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.div.glm.fit2)

#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)   0.0112343  0.0004731  23.745   <2e-16 ***
#ORP_mV      -0.0001512  0.0004968  -0.304    0.763

aug21.div.glm.fit3<-glm(formula = Bac_Shannon_Diversity ~ Temp_DegC, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.div.glm.fit3)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept) 0.0113225  0.0004451  25.439   <2e-16 ***
#Temp_DegC   0.0011947  0.0004861   2.458   0.0188 *

aug21.div.glm.fit4<-glm(formula = Bac_Shannon_Diversity ~ Dissolved_OrganicMatter_RFU, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.div.glm.fit4)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)                 0.0112493  0.0004708  23.896   <2e-16 ***
#Dissolved_OrganicMatter_RFU 0.0004269  0.0004659   0.916    0.365

aug21.div.glm.fit5<-glm(formula = Bac_Shannon_Diversity ~ Sulfate_milliM, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.div.glm.fit5)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)     0.0112325  0.0004772  23.536   <2e-16 ***
#Sulfate_milliM -0.0002664  0.0004893  -0.545    0.589

aug21.div.glm.fit6<-glm(formula = Bac_Shannon_Diversity ~ Sulfide_microM, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.div.glm.fit6)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)    0.0112379  0.0004723   23.80   <2e-16 ***
#Sulfide_microM 0.0002944  0.0005160    0.57    0.572

fit1<-aov(Bac_Shannon_Diversity ~ as.factor(Depth_m), data=aug21.div)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(aug21.div$Bac_Shannon_Diversity, aug21.div$Depth_m, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit1)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
#Depth_m      7   4097   585.3   1.114   0.38
#Residuals   31  16294   525.6
Tuk1<-TukeyHSD(fit1)
Tuk1$Depth_m

#plot(Bac_Shannon_Diversity ~ Depth_m, data=aug21.div)
#abline(aov(DustComplexity ~ Elevation, data=aug21.div))

# fit.0<-aov(DustComplexity ~ as.factor(Elevation), data=aug21.div)
# summary(fit.0)
# TukeyHSD(fit.0)
# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Shannon_Diversity ~ Depth_m, data = aug21.div)
# Fligner-Killeen:med chi-squared = 4.091, df = 7, p-value = 0.7692
# Which shows that the data do not deviate significantly from homogeneity.
compare_means(Bac_Shannon_Diversity ~ Depth_m, data=aug21.div, method="anova",p.adjust.method = "bonferroni") # won't take as.factor(Elevation) as input


#### August - Species Richness ####
# August 2021
aug21.sr.glm.fit1<-glm(formula = Bac_Species_Richness ~ DO_Percent_Local, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.sr.glm.fit1)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)       0.0112706  0.0004664  24.166   <2e-16 ***
#DO_Percent_Local -0.0009547  0.0005092  -1.875   0.0687 .

aug21.sr.glm.fit2<-glm(formula = Bac_Species_Richness ~ ORP_mV, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.sr.glm.fit2)

#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)   0.0112343  0.0004731  23.745   <2e-16 ***
#ORP_mV      -0.0001512  0.0004968  -0.304    0.763

aug21.sr.glm.fit3<-glm(formula = Bac_Species_Richness ~ Temp_DegC, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.sr.glm.fit3)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept) 0.0113225  0.0004451  25.439   <2e-16 ***
#Temp_DegC   0.0011947  0.0004861   2.458   0.0188 *

aug21.sr.glm.fit4<-glm(formula = Bac_Species_Richness ~ Dissolved_OrganicMatter_RFU, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.sr.glm.fit4)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)                 0.0112493  0.0004708  23.896   <2e-16 ***
#Dissolved_OrganicMatter_RFU 0.0004269  0.0004659   0.916    0.365

aug21.sr.glm.fit5<-glm(formula = Bac_Species_Richness ~ Sulfate_milliM, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.sr.glm.fit5)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)     0.0112325  0.0004772  23.536   <2e-16 ***
#Sulfate_milliM -0.0002664  0.0004893  -0.545    0.589

aug21.sr.glm.fit6<-glm(formula = Bac_Species_Richness ~ Sulfide_microM, data=aug21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(aug21.sr.glm.fit6)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)    0.0112379  0.0004723   23.80   <2e-16 ***
#Sulfide_microM 0.0002944  0.0005160    0.57    0.572

fit1<-aov(Bac_Species_Richness ~ as.factor(Depth_m), data=aug21.div)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(aug21.div$Bac_Species_Richness, aug21.div$Depth_m, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit1)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
#Depth_m      7   4097   585.3   1.114   0.38
#Residuals   31  16294   525.6
Tuk1<-TukeyHSD(fit1)
Tuk1$Depth_m

#plot(Bac_Species_Richness ~ Depth_m, data=aug21.div)
#abline(aov(DustComplexity ~ Elevation, data=aug21.div))

# fit.0<-aov(DustComplexity ~ as.factor(Elevation), data=aug21.div)
# summary(fit.0)
# TukeyHSD(fit.0)
# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Species_Richness ~ Depth_m, data = aug21.div)
# Fligner-Killeen:med chi-squared = 4.091, df = 7, p-value = 0.7692
# Which shows that the data do not deviate significantly from homogeneity.
compare_means(Bac_Species_Richness ~ Depth_m, data=aug21.div, method="anova",p.adjust.method = "bonferroni") # won't take as.factor(Elevation) as input


#### December - Shannon Diversity ####
# December 2021
dec21.div.glm.fit1<-glm(formula = Bac_Shannon_Diversity ~ DO_Percent_Local, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.div.glm.fit1)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)       0.0112706  0.0004664  24.166   <2e-16 ***
#DO_Percent_Local -0.0009547  0.0005092  -1.875   0.0687 .

dec21.div.glm.fit2<-glm(formula = Bac_Shannon_Diversity ~ ORP_mV, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.div.glm.fit2)

#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)   0.0112343  0.0004731  23.745   <2e-16 ***
#ORP_mV      -0.0001512  0.0004968  -0.304    0.763

dec21.div.glm.fit3<-glm(formula = Bac_Shannon_Diversity ~ Temp_DegC, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.div.glm.fit3)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept) 0.0113225  0.0004451  25.439   <2e-16 ***
#Temp_DegC   0.0011947  0.0004861   2.458   0.0188 *

dec21.div.glm.fit4<-glm(formula = Bac_Shannon_Diversity ~ Dissolved_OrganicMatter_RFU, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.div.glm.fit4)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)                 0.0112493  0.0004708  23.896   <2e-16 ***
#Dissolved_OrganicMatter_RFU 0.0004269  0.0004659   0.916    0.365

dec21.div.glm.fit5<-glm(formula = Bac_Shannon_Diversity ~ Sulfate_milliM, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.div.glm.fit5)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)     0.0112325  0.0004772  23.536   <2e-16 ***
#Sulfate_milliM -0.0002664  0.0004893  -0.545    0.589

dec21.div.glm.fit6<-glm(formula = Bac_Shannon_Diversity ~ Sulfide_microM, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.div.glm.fit6)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)    0.0112379  0.0004723   23.80   <2e-16 ***
#Sulfide_microM 0.0002944  0.0005160    0.57    0.572

fit1<-aov(Bac_Shannon_Diversity ~ as.factor(Depth_m), data=dec21.div)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(dec21.div$Bac_Shannon_Diversity, dec21.div$Depth_m, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit1)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
#Depth_m      7   4097   585.3   1.114   0.38
#Residuals   31  16294   525.6
Tuk1<-TukeyHSD(fit1)
Tuk1$Depth_m

#plot(Bac_Shannon_Diversity ~ Depth_m, data=dec21.div)
#abline(aov(DustComplexity ~ Elevation, data=dec21.div))

# fit.0<-aov(DustComplexity ~ as.factor(Elevation), data=dec21.div)
# summary(fit.0)
# TukeyHSD(fit.0)
# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Shannon_Diversity ~ Depth_m, data = dec21.div)
# Fligner-Killeen:med chi-squared = 4.091, df = 7, p-value = 0.7692
# Which shows that the data do not deviate significantly from homogeneity.
compare_means(Bac_Shannon_Diversity ~ Depth_m, data=dec21.div, method="anova",p.adjust.method = "bonferroni") # won't take as.factor(Elevation) as input

#### December - Species Richness ####
# December 2021
dec21.sr.glm.fit1<-glm(formula = Bac_Species_Richness ~ DO_Percent_Local, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.sr.glm.fit1)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)       0.0112706  0.0004664  24.166   <2e-16 ***
#DO_Percent_Local -0.0009547  0.0005092  -1.875   0.0687 .

dec21.sr.glm.fit2<-glm(formula = Bac_Species_Richness ~ ORP_mV, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.sr.glm.fit2)

#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)   0.0112343  0.0004731  23.745   <2e-16 ***
#ORP_mV      -0.0001512  0.0004968  -0.304    0.763

dec21.sr.glm.fit3<-glm(formula = Bac_Species_Richness ~ Temp_DegC, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.sr.glm.fit3)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept) 0.0113225  0.0004451  25.439   <2e-16 ***
#Temp_DegC   0.0011947  0.0004861   2.458   0.0188 *

dec21.sr.glm.fit4<-glm(formula = Bac_Species_Richness ~ Dissolved_OrganicMatter_RFU, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.sr.glm.fit4)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)                 0.0112493  0.0004708  23.896   <2e-16 ***
#Dissolved_OrganicMatter_RFU 0.0004269  0.0004659   0.916    0.365

dec21.sr.glm.fit5<-glm(formula = Bac_Species_Richness ~ Sulfate_milliM, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.sr.glm.fit5)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)     0.0112325  0.0004772  23.536   <2e-16 ***
#Sulfate_milliM -0.0002664  0.0004893  -0.545    0.589

dec21.sr.glm.fit6<-glm(formula = Bac_Species_Richness ~ Sulfide_microM, data=dec21.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(dec21.sr.glm.fit6)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)    0.0112379  0.0004723   23.80   <2e-16 ***
#Sulfide_microM 0.0002944  0.0005160    0.57    0.572

fit1<-aov(Bac_Species_Richness ~ as.factor(Depth_m), data=dec21.div)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(dec21.div$Bac_Species_Richness, dec21.div$Depth_m, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit1)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
#Depth_m      7   4097   585.3   1.114   0.38
#Residuals   31  16294   525.6
Tuk1<-TukeyHSD(fit1)
Tuk1$Depth_m

#plot(Bac_Species_Richness ~ Depth_m, data=dec21.div)
#abline(aov(DustComplexity ~ Elevation, data=dec21.div))

# fit.0<-aov(DustComplexity ~ as.factor(Elevation), data=dec21.div)
# summary(fit.0)
# TukeyHSD(fit.0)
# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Species_Richness ~ Depth_m, data = dec21.div)
# Fligner-Killeen:med chi-squared = 4.091, df = 7, p-value = 0.7692
# Which shows that the data do not deviate significantly from homogeneity.
compare_means(Bac_Species_Richness ~ Depth_m, data=dec21.div, method="anova",p.adjust.method = "bonferroni") # won't take as.factor(Elevation) as input




#### April - Shannon Diversity ####
# April 2022
apr22.div.glm.fit1<-glm(formula = Bac_Shannon_Diversity ~ DO_Percent_Local, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.div.glm.fit1)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)       0.0112706  0.0004664  24.166   <2e-16 ***
#DO_Percent_Local -0.0009547  0.0005092  -1.875   0.0687 .

apr22.div.glm.fit2<-glm(formula = Bac_Shannon_Diversity ~ ORP_mV, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.div.glm.fit2)

#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)   0.0112343  0.0004731  23.745   <2e-16 ***
#ORP_mV      -0.0001512  0.0004968  -0.304    0.763

apr22.div.glm.fit3<-glm(formula = Bac_Shannon_Diversity ~ Temp_DegC, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.div.glm.fit3)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept) 0.0113225  0.0004451  25.439   <2e-16 ***
#Temp_DegC   0.0011947  0.0004861   2.458   0.0188 *

apr22.div.glm.fit4<-glm(formula = Bac_Shannon_Diversity ~ Dissolved_OrganicMatter_RFU, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.div.glm.fit4)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)                 0.0112493  0.0004708  23.896   <2e-16 ***
#Dissolved_OrganicMatter_RFU 0.0004269  0.0004659   0.916    0.365

apr22.div.glm.fit5<-glm(formula = Bac_Shannon_Diversity ~ Sulfate_milliM, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.div.glm.fit5)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)     0.0112325  0.0004772  23.536   <2e-16 ***
#Sulfate_milliM -0.0002664  0.0004893  -0.545    0.589

apr22.div.glm.fit6<-glm(formula = Bac_Shannon_Diversity ~ Sulfide_microM, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.div.glm.fit6)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)    0.0112379  0.0004723   23.80   <2e-16 ***
#Sulfide_microM 0.0002944  0.0005160    0.57    0.572

fit1<-aov(Bac_Shannon_Diversity ~ as.factor(Depth_m), data=apr22.div)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(apr22.div$Bac_Shannon_Diversity, apr22.div$Depth_m, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit1)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
#Depth_m      7   4097   585.3   1.114   0.38
#Residuals   31  16294   525.6
Tuk1<-TukeyHSD(fit1)
Tuk1$Depth_m

#plot(Bac_Shannon_Diversity ~ Depth_m, data=apr22.div)
#abline(aov(DustComplexity ~ Elevation, data=apr22.div))

# fit.0<-aov(DustComplexity ~ as.factor(Elevation), data=apr22.div)
# summary(fit.0)
# TukeyHSD(fit.0)
# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Shannon_Diversity ~ Depth_m, data = apr22.div)
# Fligner-Killeen:med chi-squared = 4.091, df = 7, p-value = 0.7692
# Which shows that the data do not deviate significantly from homogeneity.
compare_means(Bac_Shannon_Diversity ~ Depth_m, data=apr22.div, method="anova",p.adjust.method = "bonferroni") # won't take as.factor(Elevation) as input



#### April - Species Richness ####
# April 2022
apr22.sr.glm.fit1<-glm(formula = Bac_Species_Richness ~ DO_Percent_Local, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.sr.glm.fit1)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)       0.0112706  0.0004664  24.166   <2e-16 ***
#DO_Percent_Local -0.0009547  0.0005092  -1.875   0.0687 .

apr22.sr.glm.fit2<-glm(formula = Bac_Species_Richness ~ ORP_mV, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.sr.glm.fit2)

#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)   0.0112343  0.0004731  23.745   <2e-16 ***
#ORP_mV      -0.0001512  0.0004968  -0.304    0.763

apr22.sr.glm.fit3<-glm(formula = Bac_Species_Richness ~ Temp_DegC, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.sr.glm.fit3)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept) 0.0113225  0.0004451  25.439   <2e-16 ***
#Temp_DegC   0.0011947  0.0004861   2.458   0.0188 *

apr22.sr.glm.fit4<-glm(formula = Bac_Species_Richness ~ Dissolved_OrganicMatter_RFU, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.sr.glm.fit4)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)                 0.0112493  0.0004708  23.896   <2e-16 ***
#Dissolved_OrganicMatter_RFU 0.0004269  0.0004659   0.916    0.365

apr22.sr.glm.fit5<-glm(formula = Bac_Species_Richness ~ Sulfate_milliM, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.sr.glm.fit5)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)     0.0112325  0.0004772  23.536   <2e-16 ***
#Sulfate_milliM -0.0002664  0.0004893  -0.545    0.589

apr22.sr.glm.fit6<-glm(formula = Bac_Species_Richness ~ Sulfide_microM, data=apr22.div)%>%
  adjust_pvalue(method="bonferroni")
## ^ went with linear regression because Shannon diversity and dust complexity are continuous data, despite not being normally distributed
# model form is response ~ terms (y ~ x) where response is the (numeric) response vector and terms is a series of terms which specifies a linear predictor for response.
summary(apr22.sr.glm.fit6)
#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)
#(Intercept)    0.0112379  0.0004723   23.80   <2e-16 ***
#Sulfide_microM 0.0002944  0.0005160    0.57    0.572

fit1<-aov(Bac_Species_Richness ~ as.factor(Depth_m), data=apr22.div)
# ANOVA is basically a regression but w/ categorical variables more info here https://www.statology.org/anova-vs-regression/
#pairwise.adonis(apr22.div$Bac_Species_Richness, apr22.div$Depth_m, p.adjust.m='bonferroni') # shows us variation for each sample to see which ones are different

summary(fit1)
#Df           Sum Sq Mean Sq    F value   Pr(>F)
#Depth_m      7   4097   585.3   1.114   0.38
#Residuals   31  16294   525.6
Tuk1<-TukeyHSD(fit1)
Tuk1$Depth_m

#plot(Bac_Species_Richness ~ Depth_m, data=apr22.div)
#abline(aov(DustComplexity ~ Elevation, data=apr22.div))

# fit.0<-aov(DustComplexity ~ as.factor(Elevation), data=apr22.div)
# summary(fit.0)
# TukeyHSD(fit.0)
# Levene's test with one independent variable
## Levene's tests whether variances of 2 samples are equal
## we want variances to be the same -- want NON SIGNIFICANCE!
## t test assumes that variances are the same, so Levene's test needs to be non significant
## Fligner's test is a Levene's test for data that are not normally distributed
## more here: https://www.geeksforgeeks.org/fligner-killeen-test-in-r-programming/
fligner.test(Bac_Species_Richness ~ Depth_m, data = apr22.div)
# Fligner-Killeen:med chi-squared = 4.091, df = 7, p-value = 0.7692
# Which shows that the data do not deviate significantly from homogeneity.
compare_means(Bac_Species_Richness ~ Depth_m, data=apr22.div, method="anova",p.adjust.method = "bonferroni") # won't take as.factor(Elevation) as input




#### Visualize Richness, Diversity vs Env Variables ####

## Shannon Diversity & Environmental Variables
# note: R (correlation coefficient) vs R^2 (coefficient of determination): https://towardsdatascience.com/r%C2%B2-or-r%C2%B2-when-to-use-what-4968eee68ed3

ggplot(bac.div.metadat.rar, aes(x = DO_Percent_Local, y = Bac_Shannon_Diversity)) +
  geom_point(aes(color=as.numeric(Depth_m),shape=SampDate), size=3) + theme_classic() +
  stat_smooth(method = "glm", col = "black", se=FALSE, size=1)+ labs(title="Dissolved Oxygen x 16S Shannon Diversity", color="Depth (m)")+ylab("Shannon Diversity")+xlab("Dissolved Oxygen (%)")+
  scale_colour_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) +
  scale_shape_discrete(labels=c("June 2021","August 2021","December 2021","April 2022"),name="Sample Date") +
  theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),legend.title.align=0.5, legend.title = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1),legend.text = element_text(size=11)) +
  stat_cor(label.y = 150, label.x=3) +
  stat_regline_equation(aes(label=paste(..eq.label.., ..adj.rr.label.., sep = "~~~~")),label.y = 160,label.x=3)

ggplot(bac.div.metadat.rar, aes(x = DO_Percent_Local, y = Bac_Shannon_Diversity)) +
  geom_point(aes(color=as.numeric(Depth_m),shape=SampDate), size=3) + theme_classic() +
  labs(title="Dissolved Oxygen x 16S Shannon Diversity", color="Depth (m)")+ylab("Shannon Diversity")+xlab("Dissolved Oxygen (%)")+
  scale_colour_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) +
  scale_shape_discrete(labels=c("June 2021","August 2021","December 2021","April 2022"),name="Sample Date") +
  theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),legend.title.align=0.5, legend.title = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1),legend.text = element_text(size=11))

ggplot(bac.div.metadat.rar, aes(x = ORP_mV, y = Bac_Shannon_Diversity)) +
  geom_point(aes(color=as.numeric(Depth_m),shape=SampDate), size=3) + theme_classic() +
  stat_smooth(method = "glm", col = "black", se=FALSE, size=1)+ labs(title="Oxidation-Reduction Potential x 16S Shannon Diversity", color="Depth (m)")+ylab("Shannon Diversity")+xlab("Redox Potential (mV)")+
  scale_colour_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) +
  scale_shape_discrete(labels=c("June 2021","August 2021","December 2021","April 2022"),name="Sample Date") +
  theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),legend.title.align=0.5, legend.title = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1),legend.text = element_text(size=11)) +
  stat_cor(label.y = 3, label.x=1) +
  stat_regline_equation(aes(label=paste(..eq.label.., ..adj.rr.label.., sep = "~~~~")),label.y = 3.2,label.x=1)

ggplot(bac.div.metadat.rar, aes(x = Temp_DegC, y = Bac_Shannon_Diversity)) +
  geom_point(aes(color=as.numeric(Depth_m),shape=SampDate), size=3) + theme_classic() + labs(title="Temperature x 16S Shannon Diversity", color="Depth (m)")+ylab("16S Shannon Diversity")+xlab("Temperature (C)")+
  scale_colour_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) +
  scale_shape_discrete(labels=c("June 2021","August 2021","December 2021","April 2022"),name="Sample Date") +
  theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),legend.title.align=0.5, legend.title = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1),legend.text = element_text(size=11))

## Species Richness & Environmental Variables
# note: R (correlation coefficient) vs R^2 (coefficient of determination): https://towardsdatascience.com/r%C2%B2-or-r%C2%B2-when-to-use-what-4968eee68ed3

ggplot(bac.div.metadat.rar, aes(x = DO_Percent_Local, y = Bac_Species_Richness)) +
  geom_point(aes(color=as.numeric(Depth_m),shape=SampDate), size=3) + theme_classic() +
  stat_smooth(method = "glm", col = "black", se=FALSE, size=1)+ labs(title="Dissolved Oxygen x 16S Species Richness", color="Depth (m)")+ylab("Species Richness")+xlab("Dissolved Oxygen (%)")+
  scale_colour_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) +
  scale_shape_discrete(labels=c("June 2021","August 2021","December 2021","April 2022"),name="Sample Date") +
  theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),legend.title.align=0.5, legend.title = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1),legend.text = element_text(size=11)) +
  stat_cor(label.y = 3, label.x=1) +
  stat_regline_equation(aes(label=paste(..eq.label.., ..adj.rr.label.., sep = "~~~~")),label.y = 3.1,label.x=1)

ggplot(bac.div.metadat.rar, aes(x = ORP_mV, y = Bac_Species_Richness)) +
  geom_point(aes(color=as.numeric(Depth_m),shape=SampDate), size=3) + theme_classic() +
  stat_smooth(method = "glm", col = "black", se=FALSE, size=1)+ labs(title="Oxidation-Reduction Potential x 16S Species Richness", color="Depth (m)")+ylab("Species Richness")+xlab("Redox Potential (mV)")+
  scale_colour_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) +
  scale_shape_discrete(labels=c("June 2021","August 2021","December 2021","April 2022"),name="Sample Date") +
  theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),legend.title.align=0.5, legend.title = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1),legend.text = element_text(size=11)) +
  stat_cor(label.y = 3, label.x=1) +
  stat_regline_equation(aes(label=paste(..eq.label.., ..adj.rr.label.., sep = "~~~~")),label.y = 3.2,label.x=1)

ggplot(bac.div.metadat.rar, aes(x = Temp_DegC, y = Bac_Species_Richness)) +
  geom_point(aes(color=as.numeric(Depth_m),shape=SampDate), size=3) + theme_classic() + labs(title="Temperature x 16S Species Richness", color="Depth (m)")+ylab("16S Species Richness")+xlab("Temperature (C)")+
  scale_colour_gradient(low="red",high="blue",guide = guide_colourbar(reverse = TRUE)) +
  scale_shape_discrete(labels=c("June 2021","August 2021","December 2021","April 2022"),name="Sample Date") +
  theme(axis.title.x = element_text(size=13),axis.title.y = element_text(size=13),legend.title.align=0.5, legend.title = element_text(size=13),axis.text = element_text(size=11),axis.text.x = element_text(vjust=1),legend.text = element_text(size=11))


#### Save Everything ####
save.image("data/SSeawater_AlphaDiv_Data_Rarefied.Rdata")
